### ISSUE TITLE
Basic auth can only use Attributes not "Username", I need to use "Username"

### ISSUE THREAD

**Issue by ne0YT** [goauthentik/authentik#6596]
**Describe the bug**
I can use Attributes successfully. Also the standard (when I set nothing) it will use the email-address.
but it is impossible to use the username currently.

**To Reproduce**

in Update Proxy Provider I tried setting:
```
Username
username
user
authentik-username
X-authentik-username
```

nothing works.
but Attributes like "LastName" or "ldap_uniq" are working.

**Expected behavior**
when I put in "Username" it will use the username not the email.

**Version and Deployment (please complete the following information):**
-   authentik version: [e.g. 2023.6.1]
-   Deployment: docker-compose

**Comment by ne0YT** [comment]
it is so weird!
It took a whole day and now it can use the username smh!
but when I was switching what to use before the change was immidiate for attributes or the email.

**Comment by ne0YT** [comment]
it's definitely a bug..
th esam eproblem occurs when I create a 2nd proxy provder with the same config.. even though I started directly by using "username" but it keeps sending the email

**Comment by ne0YT** [comment]
I setup authentik a few times now.. it's happening every time.

**Comment by ne0YT** [comment]
To fix the issue with sending the username instead of the password I tried the following;
(The username was always shown when checking the user-object) but not in the attributes field below but just under username.

so I added a LDAP Property Mapping like this:
Object field = attributes.username
Expression = return ldap.get('uid') 
in the Proxy Provider I enabled:
Intercept header authentication
&
Send HTTP-Basic Authentication

while sending HTTP-Basic Username Key:
I tried both "username" and "Username" = it always keeps sending the email.

for the "HTTP-Basic Password Key", I created a custom attribute in a group where the user resides in. = this works.
I even created a scope which tranlsates a request for an email-address to return the Username, which is enabled in this Proxy Provider = doesn't help.
here I tried with "email", "username" and "Username" too.

the same scope works for an OAUTH provider to return the username instead of the email
